spec/models/agents/twitter_user_agent_spec.rb

```diff
@@ -42,4 +43,20 @@
     end
   end
 
+  describe "#check that if choose time line is false then username is required" do
```

Comment:
I'd suggest leaving `:username => "tectonic",` in at the top of the spec and removing `:choose_home_time_line => 'true'`, then making a new `context` here for testing `choose_home_time_line`.

```ruby
it 'requires username unless choose_home_time_line is true' do
  expect(@checker).to be_valid

  @checker.options['username'] = nil
  expect(@checker).to_not be_valid

  @checker.options['choose_home_time_line'] = 'true'
  expect(@checker).to be_valid
end

context "when choose_home_time_line is true" do
  before do
    @checker.options['choose_home_time_line'] = true
    @checker.options.delete('username')
    @checker.save
  end

  it 'checks the home timeline' do
    ...
  end
end
```